### DISCUSSION TITLE
Re-enable OpenAndroMaps POI

### DISCUSSION THREAD

**Discussion by kds112** [devemux86/cruiser#356] [Q&A] [answered]
In the OAM forum it said that OAM has switched to POI v3. So, can you re-enable the download of these POI files? Thanks a lot!

**Comment by devemux86** [comment]
@kds112 

I do not see the OpenAndroMaps POI updated to v3 format.

I downloaded Andorra and its poi has not the new format.
(file date 2023-10-03)

**Reply by kds112** [reply]
Okay, then we have to wait... I hope there will be a positive answer from the OAM admins... 🤔

**Comment by kds112** [comment]
Hi Emux, in the OAM forum the POIs v3 are available for testing. Please can you check them and give Christian a feedback? Thanks a lot, I am looking forward to switching to v3 😊👍

**Reply by devemux86** [reply]
@kds112 thanks for the info!

OpenAndroMaps POI v3 samples seem ok. I also answered in OpenAndroMaps forum.

When they will include them in the map zips, I can enable them again in Cruiser.

**Comment by devemux86** [comment]
It seems the OpenAndroMaps POI do not contain speed cameras.

If you need speed cameras with OpenAndroMaps POI, please ask about them in OpenAndroMaps forum:
https://www.openandromaps.org/en/oam-forums

**Reply by kds112** [reply]
In Germany Navigation software with speed camera warning is illegal, and for hiking I don't need it anyway... 😂

**Reply by Leeb64** [reply]
The speed camera databases are not helpful anyway. Most of them are totally outdated and incorrect.

**Reply by devemux86** [reply]
Speed camera data comes from the OpenStreetMap database.

You can disable offline POI search and use online search,
then the POI will be the most updated from OpenStreetMap at the _time of the search_.

**Comment by kds112** [comment]
In OAM the POIs v3 will be available with the December update (see forum comment) 👍 Please enable the download before Christmas 😉

**Comment by kds112** [comment]
Will you publish a new release soon or will you "wait" till Christmas in order to include the OAM POI download? The next OAM map update with POI v3 will be there around Christmas (see OAM Website) 🎄

**Reply by devemux86** [reply]
@kds112

It is important to test them first inside Cruiser, so we wait for their update.
Then I can enable them on a new Cruiser version.

(their last update is still from November)

**Comment by devemux86** [comment]
OpenAndroMaps maps started updating with **POI v3**.
They will be available again in the next Cruiser update:

<img src="https://github.com/devemux86/cruiser/assets/3484020/a1876672-c467-4d30-a845-b3a728a01f91" width="50%"/>

**Reply by kds112** [reply]
Thanks a lot!! It will beta13, won't be? 👍🎄🏁

**Reply by devemux86** [reply]
@kds112 it will be in Beta 13.

**Comment by devemux86** [comment]
OpenAndroMaps POI are enabled again in the latest:
- https://github.com/devemux86/cruiser/discussions/390

**Reply by kds112** [reply]
It works 👍 Thanks a lot and Merry Christmas 🎄💫

**Reply by Ciutnik** [reply]
Thanks! 👍 But why is the size of the downloadable POI file the same as the maps (e.g. for Poland 1.6 GB)? Do POIs have to be downloaded together with the map? I've already downloaded the map before.

**Reply by devemux86** [reply]
OpenAndroMaps provide one zip with both map + poi inside.
Now they are downloaded one time and unzipped together.

With the internal BRouter, we have to rethink the downloads in Cruiser and simplify the processes more.

**Reply by Ciutnik** [reply]
Oh, I understand. Thanks for the clarification. This is not a problem because everyone wants to update the set: map + POI at the same time 🙂

**Reply by kds112** [reply]
Maybe you could integrate the POI download in the map download and delete the POI dialog completely (or enable it with the advanced download menu). Another simplification for normal users 🙃 If you download or activate a map, the POIs are handled automatically.

**Reply by kds112** [reply]
It would work with maps from Mapforge and OAM, but what about Freizeitkarte? Does it support POIs? I don't use this map source....

**Reply by devemux86** [reply]
Freizeitkarte does not provide POI, indeed this is a thing to consider.

We could have a different menu (basic / extended) but that also adds maintenance work.

Average users just expect to click maps for download and get everything: map + poi + routing.
This already happens with Mapsforge maps: their poi + routing data are downloaded automatically.

But the many menus may seem confusing to some users.

**Reply by kds112** [reply]
I would propose to simplify it as much as possible. Just downloading, activating, deleting etc the desired maps. And POIs, routing Information etc. are hidden and handled in the background. No additional (advanced) menus...   ? 🤔

**Reply by devemux86** [reply]
Also the 3 tabs of the download manager can be confusing,
so maybe we have to improve the design there too.

**Reply by kds112** [reply]
That's right. I am looking forward to your design proposal 😊

**Reply by kds112** [reply]
I open a new discussion topic for that. You can successfully close this one 🏁

**Reply by devemux86** [reply]
> You can successfully close this one 🏁

I will mark it after we make the public release.

**Comment by Ciutnik** [comment]
I would just like to remind:

> It seems the OpenAndroMaps POI do not contain speed cameras.

This is still true! Keep this in mind.
Thank you @devemux86 for pointing this out.

**Reply by devemux86** [reply]
I had suggested to OpenAndroMaps if they want to include them,
but their map focus is more for outdoor activities, not driving. :slightly_smiling_face:

Nevertheless, Cruiser will search online if something is not found offline.
So even if only use OpenAndroMaps POI, all requested POI can be found.

**Reply by Ciutnik** [reply]
But if you select OpenAndroMaps POI, the speed camera warning in the panel will not work?

**Reply by devemux86** [reply]
If there is internet when start navigation, it will work.

As it searches first in activated offline POI.
If there are no results, then it searches online.

**Reply by Ciutnik** [reply]
> If there is internet when start navigation, it will work.

Are you sure? I just checked on a short route with a speed camera along the way. In the case of POI Mapsforge, the speed camera (and gas station) appeared on the map and in the panel. But after changing to OAM, after recalculating the route and activating navigation, there was no speed camera on the map or in the panel (the gas station is still visible).
Of course, the Internet is available and navigation is in normal mode.

**Reply by devemux86** [reply]
> (the gas station is still visible)

Multiple different POI, so it considers that offline results exist because of the found fuel stations.

I will add a check in offline results, so online search will be activated for each not found POI type.

@Ciutnik thanks for the double checking! :+1: 
It will be available in the next Cruiser update.

**Reply by Ciutnik** [reply]
> I will add a check in offline results, so online search will be activated for each not found POI type.

For clarity: I wrote only about the case of display in the panel and voice information **during navigation**. I haven't checked other cases 🙂

**Reply by devemux86** [reply]
@Ciutnik we need an improvement to better merge the offline + online results in navigation.

In other cases like using the search button with free or categorized search,
it should already work properly getting online results if offline POI do not have them?

**Reply by devemux86** [reply]
e.g. activate only OpenAndroMaps POI and use the search button to search category for speed cameras.

**Reply by devemux86** [reply]
> I will add a check in offline results, so online search will be activated for each not found POI type.

Ready, it will be available in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
OpenAndroMaps POI are again available in [Cruiser 3.0.37](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
